packages/expo-sqlite/src/next/hooks.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+
+import { openDatabaseAsync, type Database } from './Database';
+import type { OpenOptions } from './NativeDatabase';
+
+export interface SQLiteProviderProps {
+  /**
+   * The name of the database file to open.
+   */
+  dbName: string;
+
+  /**
+   * Open options.
+   */
+  options?: OpenOptions;
+
+  /**
+   * The children to render.
+   */
+  children: React.ReactNode;
+
+  /**
+   * A custom initialization handler to run before rendering the children.
+   * You can use this to run database migrations or other setup tasks.
+   */
+  initHandler?: (db: Database) => Promise<void>;
+
+  /**
+   * Handle errors from SQLiteProvider.
+   * @default rethrow the error
+   */
+  errorHandler?: (error: Error) => void;
+}
+
+// Create a context for the SQLite database
+const SQLiteContext = createContext<Database | null>(null);
+
+// Create a provider component
+export function SQLiteProvider({
+  dbName,
+  options,
+  children,
+  initHandler,
+  errorHandler,
+}: SQLiteProviderProps) {
+  const [database, setDatabase] = useState<Database | null>(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    async function setup() {
+      try {
+        const db = await openDatabaseAsync(dbName, options);
+        setDatabase(db);
+        if (initHandler != null) {
+          await initHandler(db);
+        }
+        setLoading(false);
+      } catch (e) {
+        setError(e);
+      }
+    }
+
+    async function teardown() {
+      try {
+        await database?.closeAsync();
+      } catch (e) {
+        setError(e);
+      }
+    }
+
+    setup();
+
+    return () => {
+      teardown();
+    };
+  }, [dbName, options, initHandler]);
+
+  if (error != null) {
+    const handler =
+      errorHandler ??
+      ((e) => {
+        throw e;
+      });
+    handler(error);
+  }
+
+  if (loading) {
```

Comment:
I wonder should we pass `loading` back to the user and allow them to decide whether to render or not? 🤔

Reply:
that's a great feedback! i don't really want to return loading state and then returned `db` as `Nullable<Database>`. people would then need to handle null cases as you mention to me previously.

alternatively, what do you think if we add a `loadingFallback` property and let people to customize ui loading view? something like the `Suspense.fallback` property?

Reply:
Yes, passing back a non nullable db is much nicer. I think something like a fallback would be useful. Just to give users some more control and options if loading fails.